semantic-analysis/src/Analysis/Concrete.hs

```diff
@@ -64,10 +62,6 @@ instance Show Concrete where
   showsPrec p = showsPrec p . quote
 
 
-data Snoc a = Nil | Snoc a :> a
-  deriving (Foldable, Functor, Traversable)
-
-
```

Comment:
It looked like it would be important for `Snoc` to be accessible elsewhere. It turned out not to be, but there's no reason it shouldn't have its own home.